ArkKit/ark-render-kit/ArkCanvasSystem.swift

```diff
@@ -1,23 +1,26 @@
 import Foundation
 
-class ArkCanvasSystem: UpdateSystem {
+class ArkCanvasSystem<ImageEnum: ArkImageEnum>: UpdateSystem {
     var active: Bool
-    static let renderableComponentTypes: [any RenderableComponent.Type] = [
-        ButtonRenderableComponent.self,
-        JoystickRenderableComponent.self,
-        CircleRenderableComponent.self,
-        RectRenderableComponent.self,
-        PolygonRenderableComponent.self,
-        BitmapImageRenderableComponent.self,
-        CameraContainerRenderableComponent.self
-    ]
+
+    static func getRenderableComponentTypes() -> [any RenderableComponent.Type] {
+        [
+            ButtonRenderableComponent.self,
+            JoystickRenderableComponent.self,
+            CircleRenderableComponent.self,
+            RectRenderableComponent.self,
+            PolygonRenderableComponent.self,
+            BitmapImageRenderableComponent<ImageEnum>.self,
```

Comment:
Problematic line^. We can't just select all types of `BitmapImageRenderableComponent`.

Reply:
Currently, this implies that simple abstractions such as ImpactExplosionAnimation are no longer reusable due to typing issues.